### ISSUE TITLE
Support for httpcore-nio ?

### ISSUE THREAD

**Issue by aelmanaa** [microsoft/ApplicationInsights-Java#1344]
Hello,

we have started a proof of concept to evaluate Application Insights in comparison to other APM . so far so good with our java microservices which are written by our teams. However, we are also using thirdparty components (e.g.: wso2 apim & wso2 esb) which are running on jvms but using httpcore-nio: 

https://docs.wso2.com/display/ESB460/FAQ#FAQ-WhatistheNHTTPtransport?
NHTTP transport uses the Apache Http Core NIO library (http://hc.apache.org/httpcomponents-core-ga/httpcore-nio) underneath. This library provides low-level I/O handling and HTTP-level detail handling.

UNfortunately appinsights java agent doesn't seem to capture HTTP requests coming out from wso2 components. Would it be posible to support other dependecies than the ones listed here: https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent#dependencies-with-distributed-trace-propagation 

thanks

**Comment by aelmanaa** [comment]
@trask your commit fixes this issue ?

**Comment by trask** [comment]
no, sorry for the noise, that commit isn't related to this issue

**Comment by trask** [comment]
hi @aelmanaa! are you using org.apache.http.nio.client.HttpAsyncClient? if so, that should be supported

or does wso2 have its own http client implementation built on top of org.apache.http.nio primitives?
